lib/salestation/app/input_verification.rb

```diff
@@ -5,7 +5,8 @@ class App
     module InputVerification
       def verify_input(schema)
         -> (request) do
-          result = schema.call(request.input)
+          input = request.input
+          result = schema.call(input)
```

Comment:
I think it's actually a good thing that it doesn't return unvalidated input as this ensures we won't get any nasty surprises.

You can make the requester part of the schema like [we have in phone number manager](https://github.com/salemove/phone-number-manager/blame/master/lib/phone_number_manager/create_incoming_phone_number.rb#L12)

Reply:
engagement-registry even has custom logic for input verification https://github.com/salemove/engagement-registry/blame/master/lib/app/input_verification.rb so you can change it there to avoid having to change hundreds of schemas and tests right now

Reply:
I did change it on the input_verification.rb. The problem was that engagement-api had about 1000 tests fail because of this issue. 

Where it could validate some fields and then validate other fields based on a condition. [Such as this case](https://github.com/salemove/engagement-registry/blob/master/lib/app/engagement/update_engagement_request.rb#L22-L27)

Reply:
spoke IRL. Before I missed that this uses the old proc based chain which we no longer use for anything new.
So this change is fine here as the new logic doesn't use this method anyway.